receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -92,36 +98,57 @@ func (r *pReceiver) Start(_ context.Context, host component.Host) error {
 
 	allocConf := r.cfg.TargetAllocator
 	if allocConf != nil {
-		go func() {
-			// immediately sync jobs and not wait for the first tick
-			savedHash, _ := r.syncTargetAllocator(uint64(0), allocConf, baseCfg)
-			r.targetAllocatorIntervalTicker = time.NewTicker(allocConf.Interval)
-			for {
-				<-r.targetAllocatorIntervalTicker.C
-				hash, err := r.syncTargetAllocator(savedHash, allocConf, baseCfg)
-				if err != nil {
-					r.settings.Logger.Error(err.Error())
+		err = r.startTargetAllocator(allocConf, baseCfg)
+		if err != nil {
+			return err
+		}
+	}
+
+	r.loadConfigOnce.Do(func() {
+		close(r.configLoaded)
+	})
+
+	return nil
+}
+
+func (r *pReceiver) startTargetAllocator(allocConf *targetAllocator, baseCfg *config.Config) error {
+	r.settings.Logger.Info("Starting target allocator discovery")
+	// immediately sync jobs and not wait for the first tick
+	savedHash, err := r.syncTargetAllocator(uint64(0), allocConf, baseCfg)
+	if err != nil {
+		return err
+	}
+	go func() {
+		r.targetAllocatorIntervalTicker = time.NewTicker(allocConf.Interval)
```

Comment:
Can we not store targetAllocatorIntervalTicker on the pReceiver?  It only appears to be accessed in Shutdown().  Instead, we could just stop the ticker in the `case <-r.targetAllocatorStop` case.

(I believe it is possible to race on this if Start() and Shutdown() are called concurrently)